src/Compilation.zig

```diff
@@ -1653,6 +1653,8 @@ pub fn performAllTheWork(self: *Compilation) error{ TimerUnsupported, OutOfMemor
                     .error_msg = null,
                     .decl = decl,
                     .fwd_decl = fwd_decl.toManaged(module.gpa),
+                    // we don't want to emit optionals and error unions to headers since they have no ABI
+                    .typedefs = c_codegen.TypedefMap.init(std.testing.failing_allocator),
```

Comment:
I think use of failing_allocator is not the right approach here. This might drag in all manner of testing stuff. Really, what's expected here is for the code to not add anything to this array right? Should it perhaps be `undefined` or `null`?

Reply:
I meant to use `mem.failAllocator` but `undefined` works too.